zebra/zebra_vxlan.c

```diff
@@ -9095,8 +9118,16 @@ void zebra_vxlan_advertise_all_vni(ZAPI_HANDLER_ARGS)
 			   is_evpn_enabled() ? "enabled" : "disabled",
 			   flood_ctrl);
 
-	if (zvrf->advertise_all_vni == advertise)
+	/* This happens if when BGP daemons restarts */
+	if (zvrf->advertise_all_vni == advertise) {
```

Comment:
instead of special casing this would it make sense for zebra to clear advertise_all_vni when it detects bgpd going down?

Reply:
@chiragshah6 can you also take a quick look at this one?